pkg/pillar/cmd/loguploader/loguploader.go

```diff
@@ -408,6 +433,50 @@ func handleDNSDelete(ctxArg interface{}, key string, statusArg interface{}) {
 	log.Tracef("handleDNSDelete done for %s", key)
 }
 
+// periodically check if the log for current app metric-map url list
+// is still valid. if we don't have this domain anymore, remove them.
+// using subAppInstConfig deletion won't work since the entries of those
+// app can still wait for upload on the device, the action of uploading
+// to the url can happen after the app is deleted.
+func checkAppLogMetrics(ctx *loguploaderContext) {
+	var l, l3 []string
+
+	// get current configured apps
+	sub := ctx.subAppInstConfig
+	items := sub.GetAll()
+	for _, c := range items {
+		config := c.(types.AppInstanceConfig)
+		l = append(l, config.UUIDandVersion.UUID.String())
+	}
+
+	// get the url set in the log metric-map
+	l2 := zedcloud.GetAppURLset(log)
+	if len(l) >= len(l2) { // large or equal, not every app sends logs
```

Comment:
But there can still be very old items in GetURLSet, right?
This optimization might not be worth while due to that potential for confusion.

Reply:
Right. Since we only do once an hour, i don't have to do this here. will remove this check.